### DISCUSSION TITLE
Using ensemble with axondeepseg

### DISCUSSION THREAD

**Discussion by Mousa-saleh** [neuropoly/axondeepseg#549] [Help needed] [answered]
Hello all,

I trained axondeepseg with my own dataset and need to use other models as well to form ensemble learning.
The question is, how can I get the continuous output values (before prediction) to do averaging with other models.

Thanks,

**Comment by hermancollin** [comment, marked as answer]
Hi @Mousa-saleh 

If I understand correctly, you want to use the final layer's softmax output values (before binarization)?

**Reply by Mousa-saleh** [reply]
Yes, that's correct.

**Reply by hermancollin** [reply]
The axondeepseg python module supports this. In the [00-getting_started](https://github.com/neuropoly/axondeepseg/blob/master/notebooks/00-getting_started.ipynb) jupyter notebook, you can follow the instructions up until _2.2. Launch the segmentation of the image_. Here, the notebook tells us to use the `axon_segmentation()` function:

```python
prediction = axon_segmentation([path_testing], ["image.png"], path_model, config_network, resampled_resolutions=0.1, verbosity_level=3)
```
However, this function can also output the prediction with their associated probability maps. To activate this feature, you need to specify `prediction_proba_activate=True` in the function's arguments:
```python
prediction, pred_proba = axon_segmentation([path_testing], ["image.png"], path_model, config_network, resampled_resolutions=0.1, verbosity_level=3, prediction_proba_activate=True)
```
For an image of size _M x N_, the `pred_proba` variable will contain an array of size _M x N x 3_ with the probability maps for all 3 classes.

**Reply by hermancollin** [reply]
@Mousa-saleh I don't think there is a way to do this with the CLI but since you trained a model on your own data, you must already be familiar with the python module. Please tell me if this helps.

**Comment by Mousa-saleh** [comment]
Thank you very much Armand, it worked.
